### ISSUE TITLE
Can't override bean definition within a scope

### ISSUE THREAD

**Issue by kurt-streem** [InsertKoinIO/koin#645]
**Describe the bug**
Within a scope it seems to be impossible to override a bean definition. Internally the `BeanDefinition` instances for a scope are stored in a `HashSet`. When attempting to override an existing bean definition in a scope with one that is `==` (but that may have a completely different implementation), the existing bean definition is not replaced with the new bean definition. Setting `override = true` on the new bean definition makes no difference; in fact, it appears that flag is not taken into consideration at all for bean definitions within a scope.

The context of this is for testing, because I want to get an instance of a different class when testing a scoped bean.

**To Reproduce**
```
open class MyParentClass
class MySubClass1 : MyParentClass()
class MySubClass2 : MyParentClass()

scope(named("MYSCOPE")) {
    scoped<MyParentClass> {
        MySubClass1()
    }
}

// defined in a test file, intended to override the first definition of MyParentClass within the scope
scope(named("MYSCOPE")) {
    scoped<MyParentClass>(override = true) {
        MySubClass2()
    }
}
```
After both `scope(named("MYSCOPE"))` blocks are processed, the `ScopeRegistry` contains a single `ScopeDefinition` called MYSCOPE, as expected, and a `BeanDefinition` for `MyParentClass`. This `BeanDefinition` for `MyParentClass` is for the declaration in the first `scope(named("MYSCOPE"))` block, with actual class `MySubClass1`.

**Expected behavior**
The MYSCOPE `ScopeDefinition` should have the `BeanDefinition` for the `MyParentClass` declaration in the second `scope(named("MYSCOPE"))` block, with actual class `MySubClass2`.

**Why it's happening**
`ScopeRegistry` `saveDefinition(...)` is called twice, as it should be. In the first call, for the first `scope(named("MYSCOPE"))` block, no existing scope definitions are found, so one is created for MYSCOPE with the `MySubClass1` variant of `MyParentClass`. In the second pass the existing MYSCOPE `ScopeDefinition` is found, and an attempt is made to add the `MySubClass2` variant of `MyParentClass` to the `ScopeDefinition`'s `definitions` `HashSet`, but it is not actually added because the existing and new `BeanDefinition`s are `==` and that's how `HashSet` works. As mentioned, this logic doesn't take into account the fact that `override = true` is set for the second `BeanDefinition`.

Here's the relevant code in `ScopeRegistry`:

```
private fun saveDefinition(scopeSet: ScopeSet) {
    val foundScopeSet: ScopeDefinition? = definitions[scopeSet.qualifier.toString()]
    if (foundScopeSet == null) {
        definitions[scopeSet.qualifier.toString()] = scopeSet.createDefinition()
    } else {
        foundScopeSet.definitions.addAll(scopeSet.definitions)
    }
}
```

**Koin project used and used version (please complete the following information):**
`koin-core version 2.0.1`
`koin-android version 2.0.1`
`koin-test version 2.0.1`

**Comment by arnaudgiuliani** [comment]
Fixed in 2.1.0

**Comment by kurt-streem** [comment]
Thank you @arnaudgiuliani !
